templates/distribution/scripts/pre-apply.sh.tpl

```diff
@@ -190,6 +275,139 @@ deleteTracingMinioHA
+  if ! $kubectlbin get apiservice v1.monitoring.coreos.com; then
+    cat delete-dr-minio.yaml | $yqbin 'select(.apiVersion != "monitoring.coreos.com/v1")' > delete-dr-minio-filtered.yaml
+    cp delete-dr-minio-filtered.yaml delete-dr-minio.yaml 
+  fi
+{{- end }}
+  $kubectlbin delete --ignore-not-found --wait --timeout=180s -f delete-dr-minio.yaml 
+  echo "Minio on kube-system namespace deleted"
+}
+
+{{- if eq .reducers.distributionModulesDRVeleroBackend.to "externalEndpoint" }}
+
+{{- if eq .reducers.distributionModulesDRVeleroBackend.from "minio" }}
+deleteVeleroMinio
+{{- end }}
+
+{{- end }}
+
+{{- end }} # end distributionModulesDRVeleroBackend
+
+# ███    ██  ██████  ██ ███    ██ ██   ██     ████████ ██    ██ ██████  ███████ 
+# ████   ██ ██       ██ ████   ██  ██ ██         ██     ██  ██  ██   ██ ██      
+# ██ ██  ██ ██   ███ ██ ██ ██  ██   ███          ██      ████   ██████  █████   
+# ██  ██ ██ ██    ██ ██ ██  ██ ██  ██ ██         ██       ██    ██      ██      
+# ██   ████  ██████  ██ ██   ████ ██   ██        ██       ██    ██      ███████ 
+
+{{- if index .reducers "distributionModulesIngressNginxType" }}
+
+deleteNginx() {
+
+  $kustomizebin build $vendorPath/modules/ingress/katalog/nginx > delete-nginx.yaml
+  $kustomizebin build $vendorPath/modules/ingress/katalog/dual-nginx > delete-dual-nginx.yaml
+  $kustomizebin build $vendorPath/modules/ingress/katalog/external-dns/public > delete-external-dns.yaml
+  $kustomizebin build $vendorPath/modules/ingress/katalog/external-dns/private >> delete-external-dns.yaml
+  $kustomizebin build $vendorPath/modules/ingress/katalog/forecastle > delete-forecastle.yaml
+
+{{- if eq .spec.distribution.modules.monitoring.type "none" }}
+  if ! $kubectlbin get apiservice v1.monitoring.coreos.com; then
+    cat delete-nginx.yaml | $yqbin 'select(.apiVersion != "monitoring.coreos.com/v1")' > delete-nginx-filtered.yaml
+    cp delete-nginx-filtered.yaml delete-nginx.yaml
+    cat delete-dual-nginx.yaml | $yqbin 'select(.apiVersion != "monitoring.coreos.com/v1")' > delete-dual-nginx-filtered.yaml
+    cp delete-dual-nginx-filtered.yaml delete-dual-nginx.yaml
+    cat delete-external-dns.yaml | $yqbin 'select(.apiVersion != "monitoring.coreos.com/v1")' > delete-external-dns-filtered.yaml
+    cp delete-external-dns-filtered.yaml delete-external-dns.yaml
+    cat delete-forecastle.yaml | $yqbin 'select(.apiVersion != "monitoring.coreos.com/v1")' > delete-forecastle-filtered.yaml
+    cp delete-forecastle-filtered.yaml delete-forecastle.yaml
+  fi
+{{- end }}
+  $kubectlbin delete --ignore-not-found --wait --timeout=180s -f delete-external-dns.yaml
+  $kubectlbin delete --ignore-not-found --wait --timeout=180s -f delete-forecastle.yaml
+  $kubectlbin delete --ignore-not-found --wait --timeout=180s -f delete-dual-nginx.yaml
+  $kubectlbin delete --ignore-not-found --wait --timeout=180s -f delete-nginx.yaml
+  echo "nginx or dual nginx have been deleted from the cluster"
```

Comment:
If we decide to differentiate between the single or dual when deleting objects we could also update this message accordingly.